amp/src/test/java/org/digijava/kernel/validation/activity/ComponentFundingOrgRoleValidatorTest.java

```diff
@@ -0,0 +1,311 @@
+        AmpActivity activity = activityWithOneUndeclaredOrg();
+
+        TestFMService fmService = new TestFMService(ImmutableSet.of("/Activity Form/Components"));
+        APIField activityField = getMetaModel(fmService);
+
+        assertThat(getViolations(activityField, activity), emptyIterable());
+    }
+
+    @Test
+    public void testDisabledCommitments() {
+        AmpActivity activity = activityWithOneUndeclaredOrg();
+
+        TestFMService fmService = new TestFMService(
+                ImmutableSet.of("/Activity Form/Components/Component/Components Commitments/Commitment Table"));
+
+        APIField activityField = getMetaModel(fmService);
+
+        assertThat(getViolations(activityField, activity), emptyIterable());
+    }
+
+    @Test
+    public void testDisabledComponentOrg() {
+        AmpActivity activity = activityWithOneUndeclaredOrg();
+
+        TestFMService fmService = new TestFMService(ImmutableSet.of(
+                "/Activity Form/Components/Component/Components Commitments/Commitment Table/Component Organization"));
+
+        APIField activityField = getMetaModel(fmService);
+
+        assertThat(getViolations(activityField, activity), emptyIterable());
+    }
+
+    @Test
+    public void testAllTransactionTypes() {
+        HardcodedOrgs orgs = new HardcodedOrgs();
+        HardcodedRoles roles = new HardcodedRoles();
+
+        AmpComponentFunding funding1 = new AmpComponentFunding();
+        funding1.setTransactionType(Constants.COMMITMENT);
+        funding1.setReportingOrganization(orgs.getWorldBank());
+
+        AmpComponentFunding funding2 = new AmpComponentFunding();
+        funding2.setTransactionType(Constants.DISBURSEMENT);
+        funding2.setReportingOrganization(orgs.getUsaid());
+
+        AmpComponentFunding funding3 = new AmpComponentFunding();
+        funding3.setTransactionType(Constants.EXPENDITURE);
+        funding3.setReportingOrganization(orgs.getBelgium());
+
+        AmpComponent component1 = new AmpComponent();
+        component1.setFundings(ImmutableSet.of(funding1, funding2, funding3));
+
+        AmpOrgRole wbOrgRole = new AmpOrgRole();
+        wbOrgRole.setRole(roles.getDonorRole());
+        wbOrgRole.setOrganisation(orgs.getWorldBank());
+
+        AmpOrgRole usaidOrgRole = new AmpOrgRole();
+        usaidOrgRole.setRole(roles.getDonorRole());
+        usaidOrgRole.setOrganisation(orgs.getUsaid());
+
+        AmpOrgRole belgiumOrgRole = new AmpOrgRole();
+        belgiumOrgRole.setRole(roles.getDonorRole());
+        belgiumOrgRole.setOrganisation(orgs.getBelgium());
+
+        AmpActivity activity = new AmpActivity();
+        activity.setComponents(ImmutableSet.of(component1));
+        activity.setOrgrole(ImmutableSet.of(wbOrgRole, usaidOrgRole, belgiumOrgRole));
+
+        APIField activityField = getMetaModel();
+
+        assertThat(getViolations(activityField, activity), emptyIterable());
+    }
+
+    private AmpActivity activityWithOneUndeclaredOrg() {
```

Comment:
It would be nice to reduce the verbosity and the callers to know the inner details about the missing org id. With a more reusable method we can pass missing orgs ids.